client/src/dashboard/Assets.vue

```diff
@@ -265,6 +288,49 @@ function canEdit(data: AssetId | DeepReadonly<ApiAsset> | undefined, includeRoot
     }
     return true;
 }
+
+function setTitleRef(el: HTMLDivElement | null, id: AssetId) : void {
+    if (el) {
+        titleRefs.value[id] = el;
+    } else {
+        delete titleRefs.value[id];
+    }
+}
+function canRename(id: AssetId) : boolean {
```

Comment:
I think we can drop this function.

Change `currentRenameElement` to be `currentRenameAsset` instead (and thus track `AssetId` instead of `HTMLDivElement`) and we can just do `:contenteditable='currentRenameAsset === folder.id`.

The `canEdit` check will already be done when the `renameAsset` function is called and the context menu open is also guarded by the `canEdit` function.

This would also mean we actually don't need to track the `titleRefs` at all.

Having `contenteditable` be a direct reactive expression instead of a function is going to be more performant as it won't have to rerun the function on every render. So where possible that's always a benefit.

Reply:
True I could change that to currentRenameAsset, that would make sense. Getting rid of the function is also much nicer, I agree.

> This would also mean we actually don't need to track the titleRefs at all.

So the reason I added in `titleRefs` was because I needed a way to get the div element so that I could call `focus()` on it and select its contents. I didn't see a way to accomplish this without this.

Reply:
Ah I see, I missed that use.

It feels a little bit overkill to store a ref to every single asset when the chances are rather slim that we actually ever are going to interact with the majority of them.

We could store the currently selected asset (either by click or rightclick) in a variable I suppose. As we can get the element from the click/contextmenu events.

That said, it's not the end of the world either and usually you're not super long in the asset manager anyway, so we probably can get away with it.

I leave it up to you whether you want to change this or keep it, I'm fine with either decision.

Reply:
I was having issues because openContextMenu was being triggered by an event generated by the surrounding div, not the title's element itself. I found a way to retrieve the child element though, so I'd also like to get rid of `titleRefs` entirely.